Explorer les fichiers de baromètre JT afin d’identifier :

leur granularité,
leur période couverte,
les variables liées aux chaînes et aux thématiques,
leur pertinence pour l’axe “offre d’information”.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(
    "../data/raw/ina-barometre-jt-tv-donnees-quotidiennes-2000-2020-nbre-sujets-durees-202410.csv", 
    sep=";",
    encoding="latin1",
    header=None
    )
df.head()

,0,1,2,3,4,5
0,02/01/2000,France 3,NaN,Catastrophes,2,235
1,02/01/2000,France 3,NaN,Culture-loisirs,1,138
2,02/01/2000,France 3,NaN,Environnement,3,306
3,02/01/2000,France 3,NaN,International,2,241
4,02/01/2000,France 3,NaN,Société,2,160


In [3]:
df.columns = ["date", "channel_name", "empty_col", "theme", "nb_subjects", "duration"]
df.head(5)

,date,channel_name,empty_col,theme,nb_subjects,duration
0,02/01/2000,France 3,NaN,Catastrophes,2,235
1,02/01/2000,France 3,NaN,Culture-loisirs,1,138
2,02/01/2000,France 3,NaN,Environnement,3,306
3,02/01/2000,France 3,NaN,International,2,241
4,02/01/2000,France 3,NaN,Société,2,160


In [4]:
df.dtypes

date             object
channel_name     object
empty_col       float64
theme            object
nb_subjects       int64
duration          int64
dtype: object

In [5]:
df.duplicated().sum()

np.int64(0)

In [6]:
df.isna().sum()

date                 0
channel_name         0
empty_col       268424
theme                0
nb_subjects          0
duration             0
dtype: int64

In [7]:
df["date"] = pd.to_datetime(df["date"], dayfirst=True)

In [8]:
df["date"].min(), df["date"].max()

(Timestamp('2000-01-01 00:00:00'), Timestamp('2020-12-31 00:00:00'))

In [9]:
df["channel_name"].nunique()

5

In [10]:
df["channel_name"].unique()

array(['France 3', 'Arte', 'M6', 'France 2', 'TF1'], dtype=object)

In [11]:
df["theme"].nunique()

14

In [12]:
df["theme"].value_counts()

theme
Société                   33030
International             31604
Culture-loisirs           26247
Economie                  24386
Politique France          22200
Sport                     20486
Justice                   18971
Catastrophes              17739
Faits divers              15934
Environnement             15789
Santé                     15319
Sciences et techniques     9571
Education                  8917
Histoire-hommages          8231
Name: count, dtype: int64

In [13]:
df.duplicated(subset= ["date", "channel_name", "theme"]).sum()

np.int64(0)

In [14]:
df[["nb_subjects", "duration"]].describe()

,nb_subjects,duration
count,268424.000000,268424.000000
mean,2.184745,194.009098
std,1.917227,203.621299
min,1.000000,1.000000
25%,1.000000,89.000000
50%,2.000000,134.000000
75%,3.000000,242.000000
max,102.000000,16125.000000


In [15]:
df["avg_duration_per_subject"] = df["duration"] / df["nb_subjects"]
df[["avg_duration_per_subject"]].describe()

,avg_duration_per_subject
count,268424.000000
mean,88.959065
std,48.719761
min,1.000000
25%,62.500000
50%,88.000000
75%,108.000000
max,3145.000000


Voir la répartition par chaîne et par thématique, ainsi que les éventuelles redondances ou incohérences dans les données.

In [16]:
df["channel_name"].value_counts()

channel_name
TF1         65072
France 2    63189
M6          52975
France 3    52829
Arte        34359
Name: count, dtype: int64

Conclusion EDA : 
Le fichier doit être lu avec encoding="latin1" et header=None.
Il contient 268 424 lignes et 6 colonnes.
La colonne empty_col est entièrement vide et pourra être supprimée.
Les colonnes utiles sont :
date
channel_name
theme
nb_subjects
duration
La période couverte va de 2000-01-01 à 2020-12-31.
Le fichier contient 5 chaînes :
TF1
France 2
M6
France 3
Arte
Il contient 14 thématiques.
Aucun doublon n’est observé sur la combinaison date, channel_name, theme.
La granularité retenue est : une date + une chaîne + une thématique.
En moyenne, une ligne correspond à environ 2,18 sujets pour 194 secondes de durée cumulée.
La durée moyenne par sujet est d’environ 89 secondes, ce qui semble cohérent avec des sujets de JT.
Ce fichier est retenu comme source principale pour l’axe offre d’information / thématiques JT.